books/xdoc/fancy/xdoc.js

```diff
@@ -1051,184 +1051,131 @@ document.addEventListener("DOMContentLoaded", () => {
-    });
+        ta_data.push({
+            value: key,
+            nicename: xindexObj.topicName(key),
+            tokens: [xindexObj.topicRawname(key)],
+            nicelow: xindexObj.topicName(key).toLowerCase(),
+            uid: xindexObj.topicUid(key)
+        });
+    }
 
-    // Bloodhound natively lets you supply a sort function that just
-    // compares two elements.  However, that turned out to be too
-    // slow.  By using our own sorting function we can
-    //
-    //  (1) avoid having to look at the current typeahead value on
-    //      every single comparison, which was very slow when doing lots of
-    //      comparisons.
-    //
-    //  (2) pre-filter the array to make this mostly linear.
-    engine1.sorter = function(matches)
-    {
-	//console.log("My sorter called on " + matches.length + " elements.");
-	var curr = $("#jump").typeahead('val').toLowerCase();
-
-	var compare = function(a,b)
-	{
-	    // Special cases to ensure any literal matches come first, no
-	    // matter how unimportant they are. :)
-	    if (a.nicelow == curr) return -1;
-	    if (b.nicelow == curr) return 1;
-
-	    // Otherwise, put them in importance order.
-	    return a.uid - b.uid;
-	};
-
-	if (matches.length < 100) {
-	    // We can just sort it and that'll be plenty fast.
-	    return matches.sort(compare);
-	}
-
-	// Lots of elements -- do something fancier for more speed.  Since we
-	// are only going to show 20 results, it is definitely safe to throw
-	// away any entries whose UID is larger than some CUTOFF, as long as
-	// CUTOFF is bigger than at least 20 UIDs.  So, gather a reasonable
-	// subset of UIDs, sort them, and pick the 20th one.  Then throw away
-	// everything past it in a linear sweep.
-	var first_uids = [];
-	for(var i = 0; i < 100; ++i) {
-	    first_uids.push(matches[i].uid);
-	}
-	//console.log("Gathered up UIDs: ", first_uids);
-	first_uids = first_uids.sort(function(a,b){ return a-b; });
-	//console.log("Sorted UIDs:", first_uids);
-	var cutoff = first_uids[19];
-
-	var consider = [];
-	for(var i = 0;i < matches.length; ++i) {
-	    var entry = matches[i];
-	    if (entry.nicelow == curr || entry.uid < cutoff) {
-		consider.push(entry);
-	    }
-	}
-	//console.log("Down to " + consider.length + " entries to consider:", consider);
-
-	// Then we just sort whatever survived using our ordinary comparison
-	// function, which now doesn't need to consider the vast majority of
-	// the array.
-	return consider.sort(compare);
-    };
-
-    engine1.initialize();
-
-    $("#jump").typeahead({
-			     highlight: true,
-			     hint: true,
-			     autoselect: true
-			 },
-			 {
-			     name: "topics",
-			     displayKey: 'nicename',
-			     source: engine1.ttAdapter(),
-			     templates:
-			     {
-				 suggestion: jumpRender
-			     }
-			 });
+    // Custom substring matcher for infix matching with ranking and ACL2 Sources priority
+    function substringMatcher(data) {
```

Comment:
It would be nice if there was a comment here explaining the rank/sorting system.